src/azure-cli/azure/cli/command_modules/resource/commands.py

```diff
@@ -232,16 +232,16 @@ def load_command_table(self, _):
         g.custom_show_command('show', 'get_deployment_operations', client_factory=cf_deployment_operations)
 
     with self.command_group('deployment', resource_deployment_sdk, min_api='2018-05-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES) as g:
-        g.command('list', 'list_at_subscription_scope', table_transformer=transform_deployments_list)
-        g.show_command('show', 'get_at_subscription_scope')
-        g.command('delete', 'delete_at_subscription_scope', supports_no_wait=True)
+        g.custom_command('list', 'list_deployments_at_subscription_scope', table_transformer=transform_deployments_list)
+        g.custom_show_command('show', 'get_deployment_at_subscription_scope')
+        g.custom_command('delete', 'delete_deployment_at_subscription_scope', supports_no_wait=True)
         g.custom_command('validate', 'validate_arm_template_at_subscription_scope', table_transformer=deployment_validate_table_format, exception_handler=handle_template_based_exception)
         g.custom_command('create', 'deploy_arm_template_at_subscription_scope', supports_no_wait=True, validator=process_deployment_create_namespace, exception_handler=handle_template_based_exception)
         g.custom_command('export', 'export_subscription_deployment_template')
         g.wait_command('wait', getter_name='get_at_subscription_scope')
```

Comment:
What about wait command? Do you want to involve the warning message in wait command?

Reply:
sure. the only trick there is I need to make sure the warning is only logged once. I've updated it. let me know if it looks fine.

Reply:
According to your code, your warning message only be shown when the corresponding custom method is called. You can test it in your test environment and if you think it is good for you to show breaking change in this way, I can help you merge it now. NO other concern for me. 😊

Reply:
yeah I tested it and I think it worked properly now. If you look at custom.py line 753, I have flag there to indicate there's it's the first run of the method. I think we can merge it. :)

Reply:
Sounds Great!